### DISCUSSION TITLE
L10 - can no longer pluck expressions.  Is this intentional?

### DISCUSSION THREAD

**Discussion by fisharebest** [laravel/framework#46138] [General] [answered]
Until L9, you could use `->pluck(DB::raw(...))` in database queries.

e.g.

`DB::table('file_data')->groupBy('folder_name')->pluck(DB::raw('SUM(file_size)'), 'folder_name')`

Now this doesn't work, as pluck only accepts strings, and `Expresssion` no longer has a `__toString()` method.

The [upgrade guide](https://laravel.com/docs/10.x/upgrade#database-expressions) say that the change to the `Expression` class in L10

> does not affect end-user applications

and that if we do need it, we should use the following syntax instead:

`DB::table('file_data')->groupBy('folder_name')->pluck(DB::raw('SUM(file_size)')->getValue(DB::connection()->getQueryGrammar(), 'folder_name')`

This all seems very "un-eloquent".

I have used `->pluck(DB::raw(...))` in pretty much every application I've written.

Is this really an intended change?

Couldn't `->pluck()` accept `Expression` for both its's parameters?  Or am I missing something obvious?

**Comment by ahmett** [comment, marked as answer]
After some investigation and testing, I found the cause of the issue and fixed it. Please see: #46146
